### PR TITLE
Remove double implementation of registry override function

### PR DESCRIPTION
**Pull request by dermorz**
**What this PR does / why we need it**: This is a tiny refactoring follow-up PR to #8055. I expected there to be more potential for refactoring and clean up and I didn't want it in the initial PR where the fix was implemented to avoid distraction.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by vgramer — COMMENTED** [review]
/lgtm

File: pkg/resources/data.go [hunk 1]
```diff
@@ -440,7 +440,7 @@ func (d *TemplateData) KubermaticAPIImage() string {
 func (d *TemplateData) parseImage(image string) string {
 	named, _ := reference.ParseNormalizedNamed(image)
 	domain := reference.Domain(named)
-	reminder := reference.Path(named)
+	remainder := reference.Path(named)
```
Comment (xrstf) [inline pkg/resources/data.go:443]: But how will we now remember this?!

**Review by xrstf — APPROVED** [review]
/approve

File: pkg/resources/data.go (hunk 1 above)
Comment (dermorz) [reply pkg/resources/data.go:443]: This will forever *remin* a secret. 

:drum: *ba-dum-tss*

**Pull request merged**